sdk/core/Azure.Core/src/Serialization/AzureCoreExtensions.cs

```diff
@@ -40,5 +42,61 @@ public static class AzureCoreExtensions
         ///<returns>The data converted to the specified type.</returns>
         public static async ValueTask<T?> ToObjectAsync<T>(this BinaryData data, ObjectSerializer serializer, CancellationToken cancellationToken = default) =>
             (T?)await serializer.DeserializeAsync(data.ToStream(), typeof(T), cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to a Dictionary of string to object.
+        /// Each value in the key value pair will be strongly typed as an int, long, string, Guid, double or bool.
+        /// Each value can also be another Dictionary of string object representing an inner object or
+        /// a List of objects representing an array.
+        /// </summary>
+        /// <param name="data">The <see cref="BinaryData"/> instance to convert.</param>
+        /// <returns>The data converted to the Dictionary of string to object.</returns>
+        public static IDictionary<string, object?> ToDictionaryFromJson(this BinaryData data)
```

Comment:
So the desired behavior is that if the BinaryData is just a string, number, bool, or an array, you get back an empty dictionary?

Reply:
Should there be a way to detect this scenario with some sort of Try* method?

Reply:
Yes, we could throw here instead if we want, but by calling this method they are asking for us to treat it as a dictionary

Reply:
but this is a public method that anyone can use

Reply:
`InvalidOperationException` is typically a pretty good catch-all for the data not being in the right shape / state given what both .NET and many of our SDKs throw. I use that one quite a bit in several I've worked on when error conditions go beyond a simple `ArgumentException` or derivative.

Reply:
/cc @KrzysztofCwalina @tg-msft what do you think?

Reply:
IMO ArgumentException is more appropriate here.

Reply:
With an extension method, there's no "argument" that's obvious for the object being extended (the `this` parameter). The message for an `InvalidOperationException` reads, "The exception that is thrown when a method call is invalid for the object's current state.", which is exactly correct here. The data is not JSON.

Reply:
Oh good point about it being an extension method. The data is JSON, it is just not a JSON object. If the data is not JSON, we'd get an exception from the ToObjectFromJson call.

Reply:
What does JsonSerializer throw if you ask it to deserialize a JSON string into a C# model?  Should we throw that?

Reply:
System.Text.Json.JsonException

Reply:
That's what I was hoping.  I could live with either that or `InvalidOperationException`.

Reply:
I think I would prefer `InvalidOperationException` since the json might be perfectly fine but just not an object so the operation of trying to convert it to a Dictionary<string, object> was invalid not the json?  If no objections Ill move forward with that.